### PR TITLE
Upgrade of the UI to Bootstrap4

### PR DESCRIPTION
**Pull request by MaxNad**
I changed all pages to use bootstrap 4 instead of an older version and upgraded jQuery.

This allowed me to do some major cleanup. For instance, here are a few things I have done :
- Removed the administration menu from the top navigation bar. It only redirected to a page saying "Heyyyyoo. You are an Admin."
- Added the menu on the left hand side of the page for multiple pages that did not have it (Example: Administrator Specific Plugins)
- Removed a lot of code that was never called (Impossible code paths) or used
- Fixed broken form labels
- Removed unused third party JavaScript libraries
- Most pages are now using the grid system instead of tables

Since most pages are not build using tables anymore, the web interface is now responsive and fits better on different resolution screens.

I have deployed this version of Serpico for our internal use. It will allow me to test the changes before merging this pull request.

If some other people could help by testing this a bit, it would be wonderful since reviewing all of the changes will be a bit of a hassle.

### CODE REVIEW

File: routes/basic.rb [hunk 1]
```diff
@@ -54,6 +54,9 @@
     @user.save
   end
 
+  @admin = true if is_administrator?
```
Comment (BuffaloWill) [inline routes/basic.rb]: This will break a users ability to modify their consultant info.

File: routes/basic.rb [hunk 2]
```diff
@@ -84,6 +87,9 @@
 get '/reset' do
   redirect '/reports/list' unless valid_session?
 
+  @admin = true if is_administrator?
```
Comment (BuffaloWill) [inline routes/basic.rb]: Users should be able to access the pw reset functionality. Also, the plugin line should be removed.

File: routes/admin.rb [hunk 3]
```diff
@@ -623,6 +627,10 @@
     udo_template.destroy
   end
   @udos_templates = UserDefinedObjectTemplates.all
+
+  @admin = true if is_administrator?
```
Comment (BuffaloWill) [inline routes/admin.rb]: these should be moved to the top of the route if they are meant to enforce access control. I think this is taken care of at the top of the route, but I am good with a secondary check.

Comment (BuffaloWill) [reply routes/admin.rb]: Never mind =). I see it's for the template.

File: routes/basic.rb (hunk 2 above)
Comment (BuffaloWill) [reply routes/basic.rb]: Bah, this is for the template. Ignore this and previous comment. Sorry about that.

**Comment by BuffaloWill** [comment]
@MaxNad this looks awesome! Did a walk through and found no issues.

**Comment by MaxNad** [comment]
@BuffaloWill Cool. Thanks a lot.
We already started using it on our instance, I will leave it a few more days for us to find if there are any problems and merge it after.

If you have any tweeks you would like me to apply to the UI while I am at it, you can just reply to this and i will try to incorporate them. As a side note, #551 might be fixed in this PR.

**Pull request merged**